jme3-core/src/main/java/com/jme3/scene/Node.java

```diff
@@ -249,6 +251,23 @@ public void updateGeometricState() {
         if ((refreshFlags & RF_LIGHTLIST) != 0) {
             updateWorldLightList();
         }
+
+        boolean updateGlobalLights = (refreshFlags & RF_GLOBAL_LIGHTS) != 0;
+        if (updateGlobalLights){
+            refreshFlags &= ~RF_GLOBAL_LIGHTS;
+            // if root node, we collect the global lights
+            if (getParent() == null){ 
+                depthFirstTraversal(sx->{
```

Comment:
this is the traversal path, it is heavy, but only needs to happen if
1. you have global lights in the scene
2. lights need an update

Reply:
And from my reading only happens if a global light is literally removed or added (not moved, as I was initially worried might be the case)

Reply:
yes, only if added or removed